packages/simulation/src/convert.ts

```diff
@@ -0,0 +1,647 @@
+  const transitionTable: {[key: StateID]: [StateID, ReadSymbol][]} = result[0]
+  const curInitialState: StateID = result[1]
+  const curFinalStates: StateID[] = result[2]
+  dfaGraph.initialState = curInitialState
+  for (let curObject = 0; curObject < Object.keys(transitionTable).length; curObject++) {
+    dfaGraph.states[curObject] = { id: undefined, isFinal: undefined, x: 765, y: 330 }
+    dfaGraph.states[curObject].id = parseInt(Object.keys(transitionTable)[curObject])
+    if (curFinalStates.includes(dfaGraph.states[curObject].id)) {
+      dfaGraph.states[curObject].isFinal = true
+    } else {
+      dfaGraph.states[curObject].isFinal = false
+    }
+    // Add coordinates here when figured out
+  }
+  let transID = 0
+  for (let curFromState = 0; curFromState < Object.keys(transitionTable).length; curFromState++) {
+    for (let curTrans = 0; curTrans < transitionTable[Object.keys(transitionTable)[curFromState]].length; curTrans++) {
+      dfaGraph.transitions[transID] = { from: undefined, id: undefined, read: undefined, to: undefined }
+      dfaGraph.transitions[transID].from = Object.keys(transitionTable).map(Number)[curFromState]
+      dfaGraph.transitions[transID].id = transID
+      dfaGraph.transitions[transID].read = transitionTable[Object.keys(transitionTable)[curFromState]][curTrans][1]
+      dfaGraph.transitions[transID].to = transitionTable[Object.keys(transitionTable)[curFromState]][curTrans][0]
+      transID++
+    }
+  }
+
+  return dfaGraph
+}
+
+export const convertNFAtoDFA = (nfaGraph: FSAGraphIn): FSAGraphIn | DFAGraph => {
+  // Do some error checking (could add proper authentic error messaging)
+  if (!statesAndTransitionsPresent(nfaGraph)) {
+    return nfaGraph
+  } else if (!initialStateIsPresent(nfaGraph.initialState)) {
+    return nfaGraph
+  } else if (!finalStateIsPresent(nfaGraph.states)) {
+    return nfaGraph
+  } else {
+    // Remove unreachable states and check to make sure final state is still present
+    let testGraph = { ...nfaGraph }
+    testGraph = removeUnreachableNFAStates(testGraph)
+    if (!finalStateIsPresent(testGraph.states)) {
+      return nfaGraph
+    } else {
+      let dfaGraph = {
+        initialState: undefined as StateID,
+        states: [] as AutomataState[],
+        transitions: [] as AutomataTransition[]
+      }
+
+      // Sort the states so that they're ordered. This is important as the keys of the objects is used in the conversion algorithm
+      nfaGraph.states.sort((a, b) => a.id - b.id)
+      // Create a DFA from the given NFA
+      dfaGraph = createDFA(nfaGraph, dfaGraph)
+      // Remove unreachable states from this new dfaGraph
+      dfaGraph = removeUnreachableDFAStates(dfaGraph)
+      // Finally copy over the required elements from nfaGraph and return the final dfaGraph
+      const { initialState, states, transitions, ...dfaGraphCopy } = nfaGraph
+      const dfaGraphFinal = { ...dfaGraph, ...dfaGraphCopy }
+      return dfaGraphFinal
```

Comment:
```suggestion
      return { ...dfaGraph, ...dfaGraphCopy }
```
No need to make a temp